packages/manager/modules/notifications-sidebar/src/controller.js

```diff
@@ -31,18 +37,46 @@ export default class NotificationsCtrl {
     this.isLoading = true;
     this.numberOfActiveNotifications = 0;
 
+    // Will be bound to the click event on $document
+    this.readAllNotifications = () => {
+      this.toggle = false;
+
+      // Automatically set all unread messages to read
+      // When we close the notifications menu
+      this.NavbarNotifications.readAllNotifications(
+        this.getActiveNotifications(),
+      ).then(() => {
+        this.numberOfActiveNotifications = this.getActiveNotifications().length;
+        this.$rootScope.$emit(
+          'ovh::notifications::count',
+          this.numberOfActiveNotifications,
+        );
+      });
+
+      this.$document.off('click', this.readAllNotifications);
+    };
+
     this.$rootScope.$on('ovh::notifications::toggle', () => {
       this.toggle = !this.toggle;
       if (this.toggle) {
+        // Handle the click outside the menu
```

Comment:
**suggestion:** `menu` term looks like too generic. Why not prefer using `notifications menu` (already used L45)?

```suggestion
        // Handle the click outside the notifications menu
```